packages/core/src/shared/clients/s3Client.ts

```diff
@@ -709,3 +741,89 @@ async function createSdkClient(regionCode: string): Promise<S3> {
 function clearInternalBucketCache(): void {
     ;(S3.prototype as any).bucketRegionCache = {}
 }
+
+/**
+ * A URI parser that can parse out information about an S3 URI
+ * Adapted from
+ * @see https://github.com/frantz/amazon-s3-uri/
+ */
+export function parseS3Uri(uri: string): [region: string, bucket: string, key: string] {
```

Comment:
It's very important to avoid dependencies, but I'm not sure I understand the motivation for "inlining" this particular case. Was it pulling in outdated transitive packages, or some other reason?

Reply:
I don't think it was pulling in anything transitive. @hayemaxi suggested that we move away from the dependency just as a general standard I believe.